pkg/core/blockchain.go

```diff
@@ -350,31 +350,35 @@ func (bc *Blockchain) notificationDispatcher() {
 			if len(txFeed) != 0 || len(notificationFeed) != 0 || len(executionFeed) != 0 {
 				aer := event.appExecResults[0]
 				if !aer.TxHash.Equals(event.block.Hash()) {
-					panic("inconsistent application execution results")
+					panic("inconsistent application execution results: invalid hash of block executions result")
+				}
+				if len(aer.Executions) < 2 {
+					panic("inconsistent application execution results: missing postPersist block execution result")
 				}
 				for ch := range executionFeed {
 					ch <- aer
```

Comment:
There is an event sequencing problem here, onpersist is run before transactions and postpersist after, but an external observer will get postpersist execution results before transactions.